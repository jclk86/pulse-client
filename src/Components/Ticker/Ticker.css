@keyframes ticker {
  0% {
    transform: translate3d(0, 0, 0);
  }
  100% {
    transform: translate3d(-100%, 0, 0);
  }
}
.tcontainer {
  width: 100%;
  overflow: hidden;
}
.ticker-wrap {
  width: 100%;
  padding-left: 100%;
  background-color: black;
  margin-bottom: 30px;
  margin-top: 20px;
}
.ticker-move {
  display: inline-block;
  white-space: nowrap;
  padding-right: 100%;
  animation-iteration-count: infinite;
  animation-timing-function: linear;
  animation-name: ticker;
  animation-duration: 90s;
  animation-delay: -2s;
  padding-top: 10px;
  padding-bottom: 10px;
}
.ticker-move:hover {
  animation-play-state: paused;
}
.ticker-item {
  display: inline-block;
  padding: 0 2rem;
  -webkit-font-smoothing: antialiased;
}

.ticker_links {
  text-decoration: none;
  font-weight: bold;
  color: white;
  font-size: 24px;
}

@media only screen and (max-width: 768px) {
  .ticker-move {
    animation-duration: 60s;
  }
}
