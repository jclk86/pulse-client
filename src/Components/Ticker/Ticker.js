import React, { Component } from "react";
import NewsApiService from "../../Services/news-api-service";
import "./Ticker.css";

class Ticker extends Component {
  state = {
    news: []
  };

  componentDidMount() {
    NewsApiService.getNews().then(news =>
      this.setState({ news: news.articles })
    );
  }

  renderNewsTitles(articles) {
    return articles.map((article, i) => (
      <div className="ticker-item" key={i}>
        <a href={article.url} className="ticker_links">
          {article.title}
        </a>
      </div>
    ));
  }

  render() {
    const { news } = this.state;
    return (
      <div className="tcontainer">
        <div className="ticker-wrap">
          <div className="ticker-move">{this.renderNewsTitles(news)}</div>
        </div>
      </div>
    );
  }
}

export default Ticker;
