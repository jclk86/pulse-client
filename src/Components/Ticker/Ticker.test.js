import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Ticker from "./Ticker";
import NewsApiService from "../../Services/news-api-service";

describe(`Ticker Component`, () => {
  it("renders Ticker component without issues", () => {
    const news = NewsApiService.getNews().then(news => {
      return news;
    });

    const wrapper = shallow(<Ticker news={news}></Ticker>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
