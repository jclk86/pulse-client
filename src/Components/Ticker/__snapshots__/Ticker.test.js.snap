// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker Component renders Ticker component without issues 1`] = `
<div
  className="tcontainer"
>
  <div
    className="ticker-wrap + ticker-wrap-lights-off"
  >
    <div
      className="ticker-move"
    />
  </div>
</div>
`;
